server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -208,8 +208,11 @@ public Query toQuery(Function input, LuceneQueryBuilder.Context context) {
         arg.accept(INNER_VISITOR, ctx);
         for (Reference reference : ctx.references()) {
             if (reference.isNullable()) {
-                // we don't count empty arrays here as we count them below explicitly if 3Vl logic is needed.
-                builder.add(IsNullPredicate.refExistsQuery(reference, context, false), BooleanClause.Occur.MUST);
+                var refExistsQuery = IsNullPredicate.refExistsQuery(reference, context, false);
+                if (refExistsQuery != null) {
+                    // we don't count empty arrays here as we count them below explicitly if 3Vl logic is needed.
```

Comment:
in `IsNullPredicate` we short circuit query to null (ie full scan) in such cases. I'm wondering whether we should do the same here. I checked locally, we return some BooleanQuery but cannot come up with some test where it can behave incorrectly.

Reply:
Looks like it's fine since `innerQuery` (cast) is already `GenericFunctionQuery`, so no chance to miss smth. Maybe can short circuit with null anyway to play it safe - you decide.